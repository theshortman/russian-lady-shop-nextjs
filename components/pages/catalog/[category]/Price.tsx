import Discount from "@/components/Discount";
import { FC } from "react";

interface PriceProps {
  price: number;
  newPrice: number;
  discount: number;
}

const Price: FC<PriceProps> = ({ price, newPrice, discount }) => {
  return (
    <div>
      <span className="me-1 badge bg-success">
        {price > newPrice ? newPrice : price}
        RUB
      </span>
      {discount > 0 && <Discount value={discount} />}
    </div>
  );
};

export default Price;
