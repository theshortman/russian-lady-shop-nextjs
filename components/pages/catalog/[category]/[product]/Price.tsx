import { FC } from "react";
import Discount from "@/components/Discount";

interface PriceProps {
  price: number;
  newPrice: number;
  discount: number;
}

const Price: FC<PriceProps> = ({ price, newPrice, discount }) => {
  if (discount > 0) {
    return (
      <>
        <div className="me-1 badge bg-secondary">{price} RUB</div>
        <Discount value={discount} />
        <div className="fs-5 fw-bold text-success">{newPrice} RUB</div>
        <div className="mb-3">
          Вы экономите{" "}
          <span className="fw-bold text-danger">{price - newPrice} RUB</span>
        </div>
      </>
    );
  }

  return <div className="fs-5 fw-bold text-success">{price} RUB</div>;
};

export default Price;
