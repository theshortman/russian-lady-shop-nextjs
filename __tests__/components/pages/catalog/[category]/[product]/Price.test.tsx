import { render } from "@testing-library/react";
import Price from "@/components/pages/catalog/[category]/[product]/Price";

describe("Product page", () => {
  test("renders Price unchanged if discount is 0", () => {
    const { container } = render(
      <Price price={100} newPrice={100} discount={0} />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders Price unchanged if discount is 30", () => {
    const { container } = render(
      <Price price={100} newPrice={70} discount={30} />
    );
    expect(container).toMatchSnapshot();
  });
});
