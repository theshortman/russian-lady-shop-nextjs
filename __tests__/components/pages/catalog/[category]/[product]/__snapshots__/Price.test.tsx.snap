// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product page renders Price unchanged if discount is 0 1`] = `
<div>
  <div
    class="fs-5 fw-bold text-success"
  >
    100
     RUB
  </div>
</div>
`;

exports[`Product page renders Price unchanged if discount is 30 1`] = `
<div>
  <div
    class="me-1 badge bg-secondary"
  >
    100
     RUB
  </div>
  <div
    class="badge bg-danger"
  >
    -
    30
    %
  </div>
  <div
    class="fs-5 fw-bold text-success"
  >
    70
     RUB
  </div>
  <div
    class="mb-3"
  >
    Вы экономите
     
    <span
      class="fw-bold text-danger"
    >
      30
       RUB
    </span>
  </div>
</div>
`;
